pos_access_right/static/src/js/pos_access_right.js

```diff
@@ -0,0 +1,142 @@
+/*
+    Copyright (C) 2016-Today: La Louve (<http://www.lalouve.net/>)
+    @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+    License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+*/
+
+
+odoo.define('pos_access_right.pos_access_right', function (require) {
+    "use strict";
+
+    var screens = require('point_of_sale.screens');
+    var chrome = require('point_of_sale.chrome');
+    var models = require('point_of_sale.models');
+    var gui = require('point_of_sale.gui');
+    var core = require('web.core');
+    var _t = core._t;
+
+/* ********************************************************
+point_of_sale.gui
+******************************************************** */
+
+    // New function 'display_access_right' to display disabled functions
+    gui.Gui.prototype.display_access_right = function(user){
+        if (user.groups_id.indexOf(this.pos.config.group_negative_qty_id[0]) != -1){
+            $('.numpad-minus').removeClass('pos-disabled-mode');
+        }
+        else{
+            $('.numpad-minus').addClass('pos-disabled-mode');
+        }
+        if (user.groups_id.indexOf(this.pos.config.group_discount_id[0]) != -1){
+            $(".mode-button[data-mode='discount']").removeClass('pos-disabled-mode');
+        }
+        else{
+            $(".mode-button[data-mode='discount']").addClass('pos-disabled-mode');
+        }
+        if (user.groups_id.indexOf(this.pos.config.group_change_unit_price_id[0]) != -1){
+            $(".mode-button[data-mode='price']").removeClass('pos-disabled-mode');
+        }
+        else{
+            $(".mode-button[data-mode='price']").addClass('pos-disabled-mode');
+        }
+    };
+
+
+/* ********************************************************
+point_of_sale.models
+******************************************************** */
+
+    // load extra data from 'pos_config' (ids of new groups)
+    models.load_fields("pos.config", "group_negative_qty_id");
+    models.load_fields("pos.config", "group_discount_id");
+    models.load_fields("pos.config", "group_change_unit_price_id");
+    models.load_fields("pos.config", "group_multi_order_id");
+    models.load_fields("pos.config", "group_delete_order_id");
+
+    // Overload 'set_cashier' function to display correctly
+    // unauthorized function after cashier changed
+    var _set_cashier_ = models.PosModel.prototype.set_cashier;
+    models.PosModel.prototype.set_cashier = function(user) {
+        if (user.group_id) {
+            this.gui.display_access_right(user);
```

Comment:
Testing in runbot, when you change the cashier, it doesn't reload the permissions so the button classes are removed or added accordingly. Isn't the object attribute `groups_id`?

Reply:
oops, yes, you right